libraries/AP_GPS/AP_GPS_config.h

```diff
@@ -12,7 +12,7 @@
 #endif
 
 #ifndef AP_GPS_DRONECAN_ENABLED
-#define AP_GPS_DRONECAN_ENABLED HAL_ENABLE_DRONECAN_DRIVERS
+#define AP_GPS_DRONECAN_ENABLED AP_GPS_BACKEND_DEFAULT_ENABLED && HAL_ENABLE_DRONECAN_DRIVERS
```

Comment:
can we run the full matrix check on this? this could break a board

Reply:
Done.  Of all of the boards that compile, all changes are explained (changed on-board hwdef)